src/read.c

```diff
@@ -6699,7 +6699,7 @@ static avifResult avifDecoderApplySampleTransform(const avifDecoder * decoder, a
         return result;
     }
 
-    for (avifBool alpha = AVIF_FALSE; alpha <= decoder->alphaPresent ? AVIF_TRUE : AVIF_FALSE; ++alpha) {
+    for (avifBool alpha = AVIF_FALSE; !alpha || decoder->alphaPresent; alpha = AVIF_TRUE) {
```

Comment:
The proposed new code is hard to understand. I suggest either:

```
    // `alpha` is used as a boolean inside the for loop.
    for (int alpha = 0; alpha <= decoder->alphaPresent; ++alpha) {
```

or

```
    const avifBool alpha[2] = { AVIF_FALSE, AVIF_TRUE };
    for (int i = 0; i <= decoder->alphaPresent; ++i) {
        // Replace `alpha` with `alpha[i]` in the loop's body
    }
```

Note: The conditional expression in the original code, `alpha <= decoder->alphaPresent ? AVIF_TRUE : AVIF_FALSE`, can be simplified to ` alpha <= decoder->alphaPresent`. This fact is used above.

Reply:
Both solutions still compare an int to a bool (alphaPresent). The solution in https://github.com/AOMediaCodec/libavif/pull/2737 is the cleanest IMHO